sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridEvent.java

```diff
@@ -30,6 +39,12 @@
     private final com.azure.messaging.eventgrid.implementation.models.EventGridEvent event;
 
     private static final ClientLogger logger = new ClientLogger(EventGridEvent.class);
+    private static final JsonSerializer DESERIALIZER = JsonSerializerProviders.createInstance();
+
+    /*
+     * Cache serialized data for getData()
+     */
+    private BinaryData binaryData;
 
     /**
      * Create a new instance of the EventGridEvent, with the given required fields.
```

Comment:
Should we do a similar update to the ctors for EventGridEvent?

Reply:
I'll change EventGridEvent separately and limit this PR to CloudEvent change.